### DISCUSSION TITLE
Upload binary file in config map

### DISCUSSION THREAD

**Discussion by FabioNappi** [mia-platform/community#309] [Q&A] [answered]
### Which section?

Console

### Which is the version?

1.87.1

### Description

Hi everybody,

I am wondering how I can upload a binary file (in my case a png image) in a microservice's config map.
I need this because I have to serve the file as a static resource via microlc.

### Actual Outcome

I can only create text file in config maps

### Expected Outcome

I can upload a binary file in a config map

**Comment by fredmaggiowski** [comment]
Hi! You have to encode it as a base64 to make it work with k8s ConfigMaps (if you use it from Console you'd have to write some code in your application to decode the base64)

If you want native support you could use the [`binaryData`](https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-object) field but not within the Console UI, you'd have to switch the service to advanced and create the CM from scratch and still providing the data as a base64.

Supporting binary data could be a nice to have feature @albertotessarotto let's try and find it a place in our roadmap, seems like a small enhancement to do

**Reply by FabioNappi** [reply]
Hi @fredmaggiowski thanks for your answer. Native support is definitely the best option, but using the advanced service configuration we lose all the automatic config the backoffice configurator produces, thus we cannot switch to that mode.

Supporting binary data would solve all the problems, so it would be a very nice to have feature.

**Reply by fredmaggiowski** [reply, marked as answer]
For the time being, as a workaround, you could mount the file as a secret. Secret values are base64-encoded and k8s already manages the decode automatically; doing so you will be able to mount the file and have it as a proper binary file in the file system.

@albertotessarotto let's track this feature in the product backlog, let's see if it get enough traction to do it; it should be easy to do
